apis/v1alpha1/httproute_types.go

```diff
@@ -534,27 +537,31 @@ type HTTPRouteForwardTo struct {
 	//
 	Port PortNumber `json:"port"`
 
-	// Weight specifies the proportion of traffic forwarded to the backend
-	// referenced by the ServiceName or BackendRef field. computed as
-	// weight/(sum of all weights in this ForwardTo list). Weight is not a
-	// percentage and the sum of weights does not need to equal 100. If only one
-	// backend is specified, 100% of the traffic is forwarded to that backend.
-	// If weight is set to 0, no traffic should be forwarded for this entry.
-	// If unspecified, weight defaults to 1.
+	// Weight specifies the proportion of HTTP requests forwarded to the backend
+	// referenced by the ServiceName or BackendRef field. This is computed as
+	// weight/(sum of all weights in this ForwardTo list). For non-zero values,
+	// there may be some epsilon from the exact proportion defined here
+	// depending on the precision an implementation supports. Weight is not a
```

Comment:
Should we consider a range that is considered core support and a range that is considered extended or implementation-specific?

Reply:
I think it's reasonable to assume that every implementation can do some basic calculations to transform a ratio into a weight that matches what is desired as closely as possible. We know that at least one implementation is limited to ~65k points of precision here. That could be a starting point, but what if another implementation was limited to 10k, or maybe just 100? I think the best we can do here is to say that this will be used to calculate a proportion that implementations can translate into weights. All implementations will be able to support all valid inputs, not all will be able to provide the same level of precision though. I don't know, open to other suggestions here.